uniswap_swap_detection/src/agent.spec.ts

```diff
@@ -0,0 +1,244 @@
+      inputs: [],
+      outputs: [token1],
+    });
+  }
+  function createGetfee() {
+    return mockProvider.addCallTo(address1, 20, iface, "fee", {
+      inputs: [],
+      outputs: [fee],
+    });
+  }
+
+  describe("token swap detection handle Transaction", () => {
+    it("returns empty findings with TestTransactionEvent", async () => {
+      txEvent = new TestTransactionEvent();
+      txEvent.setBlock(20);
+      createGetToken0();
+      createGetToken1();
+      createGetfee();
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+    it("returns findings", async () => {
+      const address =
+        "0x61D3f523cd7e93d8deF89bb5d5c4eC178f7CfE76".toLowerCase();
+      // txEvent.setBlock(20);
+      createGetToken1();
+      createGetToken0();
+      createGetfee();
+
+      txEvent = new TestTransactionEvent().addEventLog(SWAP_EVENT, address, [
+        sender,
+        recipient,
+        amount0,
+        amount1,
+        sqrtPricex96,
+        liquidity,
+        tick,
+      ]);
+
+      txEvent.setBlock(20);
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "Uniswap-1",
+          description: `A swap between ${token0} and ${token1} on UniswapV3 was detected on this pool ${address}`,
+          alertId: "FORTA-1",
+          severity: FindingSeverity.Low,
+          type: FindingType.Info,
+          protocol: "polygon",
+          metadata: {
+            sender,
+            recipient,
+            tokenIn: token0,
+            tokenOut: token1,
+            amount0: amount0.toString(),
+            amount1: amount1.toString(),
+          },
+        }),
+      ]);
+    });
+
+    it("returns empty findings if not uniswap event", async () => {
+      createGetToken0();
+      createGetToken1();
+      createGetfee();
+      const address =
+        "0x2170c741c41531aec20e7c107c24eecfdd15e69c9bb0a8dd37b1840b9e0b207b".toLowerCase();
+      txEvent = new TestTransactionEvent().addEventLog(SWAP_EVENT, address, [
+        sender,
+        recipient,
+        amount0,
+        amount1,
+        sqrtPricex96,
+        liquidity,
+        tick,
+      ]);
+
+      const findings = await handleTransaction(txEvent);
+      expect(findings).toStrictEqual([]);
+    });
+    it("returns empty findings on non swap event", async () => {
+      createGetToken0();
+      createGetToken1();
+      createGetfee();
+      const address =
+        "0x61D3f523cd7e93d8deF89bb5d5c4eC178f7CfE76".toLowerCase();
+
+      txEvent = new TestTransactionEvent().addEventLog(CREATED_POOL, address, [
+        token0,
+        token1,
+        fee,
+        tick,
+        address,
+      ]);
+
+      const findings = await handleTransaction(txEvent);
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns multiple findings for uniswap swap event", async () => {
```

Comment:
Suggestion to add events that should not be flagged, such as non-Uniswap `swap`s. As well as another event emission (the `poolCreated` you've used) from a valid Uniswap pool.

Just to make sure that the bot is detecting what it should and filtering out the rest.